macos/FluentUI/Button/Button.swift

```diff
@@ -5,158 +5,254 @@
-	///   - style: The ButtonStyle, defaulting to primaryFilled
-	@objc public init(title: String, image: NSImage, imagePosition: NSControl.ImagePosition, style: ButtonStyle = .primaryFilled) {
-		super.init(frame: .zero)
-		self.imagePosition = imagePosition
-		initialize(title: title, image: image, style: style)
+	///   - image: The NSImage to diplay in the button
+	///   - imagePosition: The position of the image relative to the title, default .imageLeading
+	///   - size: The ButtonSize, default .large
+	///   - style: The ButtonStyle, default .primary
+	///   - accentColor: The accent or highlight NSColor, defaulting to the app primary color
+	@objc public convenience init(
+		title: String,
+		image: NSImage,
+		imagePosition: NSControl.ImagePosition = .imageLeading,
+		size: ButtonSize = .large,
+		style: ButtonStyle = .primary,
+		accentColor: NSColor = Colors.primary
+	) {
+		let format = ButtonFormat(size: size, style: style, accentColor: accentColor)
+		self.init(title: title, image: image, imagePosition: imagePosition, format: format)
 	}
 
-	/// Initializes a Fluent UI Button
-	/// Set style to primaryFilled as default
-	@objc public init() {
-		super.init(frame: .zero)
-		initialize(title: nil, image: nil, style: .primaryFilled)
+	/// Initializes a Fluent UI Button with no title or image, and with default formatting
+	@objc public convenience init() {
+		self.init()
 	}
 
 	@available(*, unavailable)
 	required public init?(coder decoder: NSCoder) {
 		preconditionFailure()
 	}
-	
-	private func initialize(title: String?, image: NSImage?, style: ButtonStyle = .primaryFilled) {
-		// Do common initialization work
+
+	/// Swift-only designated initializer accepting ButtonFormat struct.
+	/// - Parameters:
+	///   - title: String displayed in the button, default nil
+	///   - image: The NSImage to diplay in the button, default nil
+	///   - imagePosition: The position of the image, relative to the title, default imageLeading
+	///   - format: The ButtonFormat including size, style and accentColor, with all applicable defaults
+	public init(
+		title: String? = nil,
+		image: NSImage? = nil,
+		imagePosition: NSControl.ImagePosition = .imageLeading,
+		format: ButtonFormat = ButtonFormat()
+	)
+	{
+		super.init(frame: .zero)
 		isBordered = false
 		wantsLayer = true
 		layer?.contentsScale = window?.backingScaleFactor ?? 1.0
 
-		self.style = style
+		if let cell = cell as? ButtonCell {
+			// The Button properties `contentTintColorDisabled`,
+			// `backgroundColorDisabled`, and `borderColorDisabled` fully
+			// specify the look of the disabled button, and may already include
+			// a 'disabled' system effect.  Don't apply the effect again on top
+			// of that.
+			cell.imageDimsWhenDisabled = false
+		}
 
 		if let title = title {
 			self.title = title
-		}
-
-		if let image = image {
+			if let image = image {
+				self.image = image
+				self.imagePosition = imagePosition
+			} else {
+				self.imagePosition = .noImage
+			}
```

Comment:
nit: Indentation looks off?

Reply:
It is correct.  There is an extra level compared to the original, because we need to distinguish the case where *both* `title` and `image` are set to determine whether or not to override `imagePosition`.